nemo/utils/data_utils.py

```diff
@@ -46,7 +47,11 @@ def is_datastore_path(path) -> bool:
     """Check if a path is from a data object store.
     Currently, only AIStore is supported.
     """
-    return path.startswith('ais://') or path.startswith('s3://')
+    try:
+        result = urlparse(path)
+        return result.scheme in ["ais", "s3"]
```

Comment:
Let's make this future-proof - right now it will break if the user provides GCP or Azure URL schema here. You can use: `return bool(result.scheme) and bool(result.netloc)` (see urlparse docs to understand why https://docs.python.org/3/library/urllib.parse.html#url-parsing )